### PR TITLE
dev-java/{eclipse-ecj,swt}: add 4.31

### PR DESCRIPTION
**Pull request by vaukai**

### CODE REVIEW

File: dev-java/eclipse-ecj/files/ecj-4.31 [hunk 1]
```diff
@@ -0,0 +1,7 @@
+JAVAC="@GENTOO_PORTAGE_EPREFIX@/usr/bin/ecj-4.31"
+PACKAGE="=dev-java/eclipse-ecj-4.31*"
+SUPPORTED_TARGET="1.1 1.2 1.3 1.4 1.5 1.6 1.7 1.8 9 10 11 12 13 14 15 16 17 18 19 20 21"
+SUPPORTED_SOURCE="1.3 1.4 1.5 1.6 1.7 1.8 9 10 11 12 13 14 15 16 17 18 19 20 21"
```
Comment (fordfrog) [inline dev-java/eclipse-ecj/files/ecj-4.31:4]: looking at the source it also provides constants for java 22 so i'll add version 22 to the file.

Comment (vaukai) [reply dev-java/eclipse-ecj/files/ecj-4.31:4]: > looking at the source it also provides constants for java 22 so i'll add version 22 to the file.

for future cases, and also still for 4.27, where exactly to look?

Comment (fordfrog) [reply dev-java/eclipse-ecj/files/ecj-4.31:4]: it should be enough to check the constants defined for that specific version:
https://github.com/eclipse-jdt/eclipse.jdt.core/blob/84ea8767656b2c49d9b1336bbc9641cd67e93a4f/org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/impl/CompilerOptions.java#L231-L254

Comment (vaukai) [reply dev-java/eclipse-ecj/files/ecj-4.31:4]: > it should be enough to check the constants defined for that specific version: https://github.com/eclipse-jdt/eclipse.jdt.core/blob/84ea8767656b2c49d9b1336bbc9641cd67e93a4f/org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/impl/CompilerOptions.java#L231-L254

thanks for the link

**Pull request closed without merging**